debug.go

```diff
@@ -228,6 +231,9 @@ func functionInfo(p Debug, f closure) (d Debug) {
 func (l *State) functionName(ci *callInfo) (name, kind string) {
```

Comment:
```suggestion
func (l *State) functionName(ci *callInfo) (name, kind string) {
	if ci == &l.baseCallInfo {
		return
	}
```

It's a little complicated, but this case is triggered by
https://github.com/Shopify/go-lua/blob/b252935ae7ee132380e874826ecfadb3b5515afe/debug.go#L364-L365
when we reach the start of the stack. The initial stack frame is "fake" - a sentinel representing the initial call from Go to Lua. We check for the sentinel when walking the stack:
https://github.com/Shopify/go-lua/blob/b252935ae7ee132380e874826ecfadb3b5515afe/debug.go#L198-L202
but we reach past it in `Info`.

Reply:
Awesome! Thanks @fbogsany. I wasn't aware of `baseCallInfo`